From a14b13a2e86da38f14defdcd38fe6dba99937a5a Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Thu, 27 Nov 2025 03:52:07 +0000
Subject: [PATCH 06/23] set LLVM_DEFAULT_TARGET_TRIPLE if specified

Set target triple when compiling the amd-llvm
if environment variable
ROCM_SDK_TARGET_TRIPLE has been set.
This fixes the building of TheRock on Mageia Linux.

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 compiler/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/compiler/CMakeLists.txt b/compiler/CMakeLists.txt
index 53042d8f..127e44ec 100644
--- a/compiler/CMakeLists.txt
+++ b/compiler/CMakeLists.txt
@@ -13,6 +13,9 @@ if(THEROCK_ENABLE_COMPILER)
   if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     list(APPEND _extra_llvm_cmake_args "-DLLVM_ENABLE_PEDANTIC=OFF")
   endif()
+  if(DEFINED ENV{ROCM_SDK_TARGET_TRIPLE})
+    list(APPEND _extra_llvm_cmake_args "-DLLVM_DEFAULT_TARGET_TRIPLE=$ENV{ROCM_SDK_TARGET_TRIPLE}")
+  endif()
 
   # Building Flang is very memory-intensive on systems with many cores. Allow
   # users to specify the level of concurrency for these tasks so they can
-- 
2.52.0

